packages/components/psammead-topic-tags/src/index.jsx

```diff
@@ -0,0 +1,130 @@
+import {
+  string,
+  element,
+  object,
+  oneOf,
+  oneOfType,
+  shape,
+  arrayOf,
+} from 'prop-types';
+import styled from '@emotion/styled';
+import { C_LUNAR, C_EBON, C_METAL } from '@bbc/psammead-styles/colours';
+import { GEL_SPACING_HLF, GEL_SPACING } from '@bbc/gel-foundations/spacings';
+import { getSansRegular } from '@bbc/psammead-styles/font-styles';
+import { getBrevier } from '@bbc/gel-foundations/typography';
+import { scriptPropType } from '@bbc/gel-foundations/prop-types';
+
+const [TAG_TEXT_PAD_Y, TAG_TEXT_PAD_X, ROW_SPACING] = [
+  '0.375rem',
+  '0.4375rem',
+  '0.0625rem',
+];
+
+const topicTagParent = `
+  word-break: break-word;
+  margin-top: ${ROW_SPACING};
+  margin-left: ${GEL_SPACING_HLF};
+  margin-right: ${GEL_SPACING_HLF};
+  a {
+    display: flex;
+    padding-top: ${GEL_SPACING};
+    padding-bottom: ${GEL_SPACING};
+    text-decoration: none;
+    color: ${C_EBON};
+
+    &:hover, &:focus {
+      text-decoration: underline;
+    }
+    &:visited {
+      color: ${C_METAL};
+    }
+  }
+
+  a > span {
+    background-color: ${C_LUNAR};
+    padding: ${TAG_TEXT_PAD_Y} ${TAG_TEXT_PAD_X};
+  }
+`;
+
+const topicTagsContainer = `
+  display: flex;
+  flex-wrap: wrap;
+  margin-top: -${ROW_SPACING};
+  margin-bottom: 0;
+  margin-left: -${GEL_SPACING_HLF};
+  margin-right: -${GEL_SPACING_HLF};
+  padding: 0;
+`;
+
+const SingleTopicTagContainer = styled.div`
+  ${topicTagsContainer}
+`;
+
+const TopicsList = styled.ul`
+  list-style-type: none;
+  ${topicTagsContainer}
+`;
+
+const SingleTopicTagItem = styled.div`
+  ${({ service }) => getSansRegular(service)}
+  ${({ script }) => script && getBrevier(script)}
+
+  ${topicTagParent}
+`;
+
+export const TopicTag = ({ topicName, topicLink }) => (
+  <a href={topicLink}>
+    <span>{topicName}</span>
+  </a>
+);
+
+export const TopicTags = ({ dir, children, script, service }) => {
+  const hasMultipleChildren = children.length > 1;
+
+  return (
+    <>
+      {hasMultipleChildren ? (
+        <TopicsList dir={dir} role="list" service={service} script={script}>
+          {children.map((child, i) => {
+            const key = i;
+            return (
+              <SingleTopicTagItem
+                as="li"
+                dir={dir}
+                key={key}
+                service={service}
+                script={script}
+              >
+                {child}
+              </SingleTopicTagItem>
+            );
+          })}
+        </TopicsList>
+      ) : (
+        <SingleTopicTagContainer dir={dir} service={service} script={script}>
+          <SingleTopicTagItem dir={dir} service={service} script={script}>
+            {children}
+          </SingleTopicTagItem>
+        </SingleTopicTagContainer>
+      )}
+    </>
+  );
+};
+
+TopicTag.propTypes = {
+  topicName: string.isRequired,
+  topicLink: string.isRequired,
+};
+
+TopicTags.propTypes = {
+  dir: oneOf(['ltr', 'rtl']),
+  children: oneOfType([arrayOf(element), object]),
```

Comment:
Should this propType restrict the `children` to be an array of `TopicTag` objects? You can use the propType above

Reply:
Wasn't aware this was possible 😮 Starting to really like TypeScript. In the case where only one `TopicTag` is given, the `children` prop is no longer an array, but an object instead. Think `oneOfType([arrayOf(TopicTag), object])` would be the way forward?

Reply:
`oneOfType([arrayOf(TopicTag)])` was what I was thinking but my proptypes knowledge is a bit ropey! Don't get hung up on this too much if you can't get it to work. propTypes are okay but a miserable substitute for real types tbh, I find them better as code metadata tham anything functional!

Reply:
That still seems to throw a console error, I think it's because when there's only 1 `TopicTag` the `children` prop isn't an array. Weirdly `oneOfType([arrayOf(TopicTag), TopicTag])` doesn't seem to work either, but `oneOfType([arrayOf(TopicTag), object])` does. I'll investigate this further on Monday to see if I can introduce a bit more specificity when there's only one `TopicTag` 👍

Reply:
ah `oneOfType([arrayOf(TopicTag), TopicTag])` _may work_, that matches either:
- an array of topic tags
- or a single topic tag
Again no need to sink to much time into this

Reply:
Thanks @andrewscfc ,
Yeah I gave that a go and would expect that to work, but it throws a type warning when rendering, I think because when only 1 child is given, it's formatted as a weird jsx object instead of a react element. I found the `objectOf` method and it seems to work in this instance, so the propType would be:
`children: oneOfType([arrayOf(TopicTag), objectOf(TopicTag)])`

Reply:
React `children` can be anything from elements to arrays. The type `node` is useful here.

If we check the type is renderable, do we really need to check it's a TopicTag? Any over-engineering we do here will limit us in future – if we decide we want a `TopicTagExpandable` for example.

Reply:
Thanks @simonsinclair, does this mean the `propType` should replace any mention of `TopicTag` with `node`?
So: `children: oneOfType([arrayOf(node), objectOf(node)])`?

Reply:
`children: node,` will work for all cases.

Reply:
Oh awesome, thanks!